python/mlc_chat/support/auto_weight.py

```diff
@@ -33,15 +33,16 @@ def detect_weight(
         Otherwise, check the weights are in that format.
         Available weight formats:
             - auto (guess the weight format)
-            - PyTorch (validate via checking pytorch_model.bin.index.json)
-            - SafeTensor (validate via checking model.safetensors.index.json)
-            - AWQ
-            - GGML/GGUF
+            - huggingface-torch (validate via checking pytorch_model.bin.index.json)
+            - huggingface-safetensor (validate via checking model.safetensors.index.json)
+            - awq
+            - ggml
+            - gguf
 
     Returns
     -------
-    weight_path : pathlib.Path
-        The path that points to the weights.
+    weight_config_path : pathlib.Path
```

Comment:
Other weight formats may not have `config.json` file, like ggml and gguf.

Reply:
I think For huggingface-torch/huggingface-safetensor, we can return a path to bin/safetensors/config.json.
For other formats, we can return a path to the weight file, like `llama-2-7b.Q4_K_M.gguf`
What do you think?

Reply:
For other formats, yes that works for me. For huggingface weights, it would be a lot more convenient for me if we can have the weight config json file like `model.safetensors.index.json` instead of the `config.json` which will be directly feed to loader.

Reply:
Yes, you are right.
For huggingface-torch/huggingface-safetensor, we should return a path to
- bin
- safetensors
- model.safetensors.index.json
- pytorch_model.bin.index.json